File: frontend/src/services/orderService.js
import axios from 'axios';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

export const orderService = {
    getAllOrders: async () => {
        const response = await axios.get(`${API_URL}/orders`);
        return response.data;
    },

    getOrderById: async (id) => {
        const response = await axios.get(`${API_URL}/orders/${id}`);
        return response.data;
    },

    getOrdersByUserId: async (userId) => {
        const response = await axios.get(`${API_URL}/orders/user/${userId}`);
        return response.data;
    },

    getOrdersByStatus: async (status) => {
        const response = await axios.get(`${API_URL}/orders/status/${status}`);
        return response.data;
    },

    createOrder: async (order) => {
        const response = await axios.post(`${API_URL}/orders`, order);
        return response.data;
    },

    updateOrderStatus: async (id, status) => {
        const response = await axios.patch(`${API_URL}/orders/${id}/status?status=${status}`);
        return response.data;
    },

    deleteOrder: async (id) => {
        await axios.delete(`${API_URL}/orders/${id}`);
    }
};

File: backend/src/main/java/com/example/service/OrderService.java
package com.example.service;

import com.example.exception.BadRequestException;
import com.example.exception.ResourceNotFoundException;
import com.example.model.Order;
import com.example.model.OrderItem;
import com.example.repository.OrderRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.math.BigDecimal;
import java.util.List;
import java.util.Optional;

@Service
public class OrderService {
    
    @Autowired
    private OrderRepository orderRepository;
    
    public List<Order> getAllOrders() {
        return orderRepository.findAll();
    }
    
    public Optional<Order> getOrderById(Long id) {
        return orderRepository.findById(id);
    }
    
    public List<Order> getOrdersByUserId(Long userId) {
        return orderRepository.findByUserId(userId);
    }
    
    public List<Order> getOrdersByStatus(Order.OrderStatus status) {
        return orderRepository.findByStatus(status);
    }
    
    @Transactional
    public Order createOrder(Order order) {
        // Validar itens
        if (order.getItems() == null || order.getItems().isEmpty()) {
            throw new BadRequestException("Order must have at least one item");
        }
        
        // Calcular total dos itens
        BigDecimal total = BigDecimal.ZERO;
        for (OrderItem item : order.getItems()) {
            item.setOrder(order);
            // Calcular subtotal já é feito no @PrePersist do OrderItem
            total = total.add(item.getPrice().multiply(BigDecimal.valueOf(item.getQuantity())));
        }
        
        order.setTotalAmount(total);
        return orderRepository.save(order);
    }
    
    public Order updateOrderStatus(Long id, Order.OrderStatus status) {
        Order order = orderRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Order", "id", id));
        
        order.setStatus(status);
        return orderRepository.save(order);
    }
    
    public void deleteOrder(Long id) {
        Order order = orderRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Order", "id", id));
        orderRepository.delete(order);
    }
}
